src/InstrumentationEngine.SiteExtensionTestHelper/scmApplicationHost.xdt

```diff
@@ -0,0 +1,32 @@
+﻿<?xml version="1.0"?>
+<!-- Copyright (c) Microsoft Corporation. All rights reserved.
+     Licensed under the MIT License. -->
+<configuration xmlns:xdt="http://schemas.microsoft.com/XML-Document-Transform">
+  <!--
+       TL;DR: Do not include <location path="..." xdt:Locator="Match(path)">
+
+       FACTS: 1) path="..." in <location /> attribute enforces this file to only transform the kudu w3wp and not site w3wp.
+              2) The site w3wp looks at applicationHost.xdt (if exists) otherwise no-op.
+              3) the kudu w3wp looks at scmApplicationHost.xdt (if exists) otherwise applicationHost.xdt (if exists) otherwise no-op.
+              4) XDTs that omit <location /> will overwrite "previous" xdts that provide it for the given section, but will merge the section with previous XDTs that also omit it.
+                 Section refers to the root element under <location />.
+                 (see https://msdn.microsoft.com/library/dd465326.aspx#Anchor_2)
+              5) Preinstalled Site Extensions apply their XDTs before Private Site Extensions, and XDTs are transformed in alphabetical order by site extension name.
+                 eg. "Preinstall Z" xdt then "Private A" xdt then "Private B" xdt
+
+       SUMMARY: The preinstalled InstrumentationEngine must omit the <location /> attribute in this file in order to not be overwritten by private
+                site extensions that do not provide an scmApplicationHost.xdt file but do provide an applicationHost.xdt file that omits <location /> and
+                provides system.webServer/runtime/environmentVaraibles. This will cause the overriding behavior for kudu environment variables.
+  -->
+  <system.webServer>
+    <runtime xdt:Transform="InsertIfMissing">
+      <environmentVariables xdt:Transform="InsertIfMissing">
+        <!-- Set the location of the InstrumentationEngine site extension just in case the private site extension is not enabled. This will allow testing to more easily
+             locate the InstrumentationEngine site extension files without having to resort to complicated logic of determining if it is preinstalled vs private, which
+             version to use, etc
+        -->
+        <add name="MicrosoftInstrumentationEngine_InstallationRoot" value="%HOME%\SiteExtensions\InstrumentationEngine" xdt:Locator="Match(name)" xdt:Transform="InsertIfMissing" />
```

Comment:
>MicrosoftInstrumentationEngine_InstallationRoot [](start = 19, length = 47)

I'm not opposed to do this, but wouldn't it be simpler to just set an AppSetting with this value instead of deploying an entire site extension in test scenarios?

Reply:
Good call; I was thinking we could use this site extension for anything else we needed to add just for testing purposes, but this would simplify it. I'll remove this site extension and use the app setting instead and we can revisit using the test site extension later if we need to add additional functionality.